Dark redesign

robalb opened the pull request:
Changes:
- dank redesign
- new pages:
  Leaderboard page
  User page
- splitted js and css in modules, shared by the different pages
- reduced the amount of api calls from index: now it doesn't update when the tab is minimized or out of focus

 Todo
Improve design for the user page, and add other features (show active / old investments, chart of the activities, user badges)

robalb committed:
initial changes on design

sidebar redesign
dark theme

robalb committed:
sidebar redesign

robalb committed:
changed text and buttons color

robalb committed:
error toast redesign and fixed issue #157

robalb committed:
misc components redesign

robalb committed:
made collapsible elements darker and cooler

also a little bit more laggy

robalb committed:
performance inprovement on api polling

now the page stops polling the apis for updates when its not visible (minimized or in another tab)  and restart as soon as it focused back. This also fixes the bug where connection error popups accumuled on the window and filled it completely when the page was not focused

robalb committed:
redesigned leaderboard table

robalb committed:
adjusted liks underline, initial chart restyle

fix chart bug(more elements on left than on right) and unstiled dark border

robalb committed:
finished redesign

robalb committed:
organized structure

robalb committed:
first change to wall

robalb committed:
added blank table

robalb committed:
first top3 cards design

robalb committed:
finished design

and it's beautiful

robalb committed:
shrinked 2 and 3 images

robalb committed:
disabled buttons

robalb committed:
moving to es6 modules

this breaks all the inline scripts, like onclick attributes on buttons.

robalb committed:
moved modules to separate files

robalb committed:
fixed modal style

robalb committed:
imported libraries for wall

robalb committed:
adding wall module

robalb committed:
Merge remote-tracking branch 'upstream/master' into dark-redesign

robalb committed:
added beta blob

robalb committed:
created code for the top 3 cards updater

robalb committed:
added loeaderboard swap animation

robalb committed:
small tweaks to mobile css

robalb committed:
added user page

robalb committed:
created user scripts, added redirect to user from searchbar

robalb committed:
changed nav links

robalb commented:
Note: this fixes #157 and part of #146 .
I have tested the website only on firefox for windows
And on chrome for android.
Explorer is not supported

rickles42 commented on docs/wall.html:
```diff
@@ -0,0 +1,225 @@
+<body>
+    <nav>
+        <div class="nav-wrapper">
+            <a href="#" class="brand-logo"><img src="resources/logo240.png" class="logo-img"/><span
```
Let's have the nav logo and name bring the user back to index.html

rickles42 commented on docs/wall.html:
```diff
@@ -0,0 +1,225 @@
+    <div class="container" >
+      <div class="section center-align">
+        <h4 class="no-margin-top" >Top Traders wall</h4>
+        <p>Select one of the market seasons below to view it's top 50 traders.</p>
```
The purpose of this page is already clear. We can simplify the instructions to "Select a season" or even remove them entirely and just put a label next to the buttons that says, "Season:"

rickles42 commented on docs/wall.html:
```diff
@@ -0,0 +1,225 @@
+
+    <div class="container" >
+      <div class="section center-align">
+        <h4 class="no-margin-top" >Top Traders wall</h4>
```
Let's be consistent in the naming of this page, and always call it "Leaderboards"

rickles42 commented on docs/wall.html:
```diff
@@ -0,0 +1,225 @@
+        </li>
+        <!-- <li><a href="foo.html">brokers</a></li> -->
+        <li><a href="index.html">home</a></li>
+        <li><a class="grey-text" >leaderboard</a></li>
```
Change nav to "leaderboards"

rickles42 commented on docs/index.html:
```diff
@@ -158,7 +165,10 @@ <h4 class="row-title">Overview</h4>
                     <div class="col s12 m4 l4">
                         <h4>Top Investors</h4>
                         <p>This is a list of the top 5 investors since 2018-06-29<br>
+                        <a href="./wall.html" class="underlined" >view the wall of fame</a>
```
Change to something like "view full leaderboards"

rickles42 commented on docs/index.html:
```diff
@@ -82,14 +89,14 @@
-                        <h4 class="row-title">Overview</h4>
+                        <h4 class="no-margin-top">Overview</h4>
                         <p>Market active since 2018-06-29</p>
                         <p class="overview-text">Active investments: <span id="active-investments">--</span>, detaining
```
Let's clean up the summary text to three lines:

"Active investments: ---"
"Total invested: --- M¢"
"Total balances: --- M¢"

rickles42 commented on docs/wall.html:
```diff
@@ -0,0 +1,225 @@
+        
+      </div>
+      
+      <div class="table-wrap" >
```
Can we clean up the sizing of the leaderboards columns? On my laptop (using Edge browser) there's a bunch of empty space to the right of "Completed investments". Looks like the table should be about 75% of its current width.

robalb replied:
Can you send me a screenshot? Just to see if the problem is related to edge, or your screen dimensions

rickles42 replied:
@robalb Sent to you on Slack.

rickles42 reviewed (changes requested):
Looks great overall, just have some minor feedback

rickles42 commented:
Also, I got a "INFO: Write your message here" popup when I first launched the page - I feel like we should just remove that bit until we want to push another message"

robalb committed:
tweaks to index

robalb committed:
tweaks to leaderboards

robalb committed:
tweaks to user

robalb committed:
disabled popup info

robalb commented:
ok, i've 
- fixed the logo that didn't link to the index
- cleaned the summary text
- fixed the naming inconsistency for the leaderboards page
- disabled the info popup

the leaderboards table width is a little bit more complicated to fix, i'll do this another time, i also have to add a limit to how much the other pages can stretch horizontally.
